sycl/test-e2e/ESIMD/unified_memory_api/Inputs/block_store.hpp

```diff
@@ -0,0 +1,178 @@
+
+           else
+             block_store(Out + ElemOff, Vals);
+
+           Vals = block_load<T, N>(Out + ElemOff);
+           Vals += 1;
+           if constexpr (UseProperties)
+             block_store(Out, ElemOff * sizeof(T), Vals, StorePropertiesT{});
+           else
+             block_store(Out, ElemOff * sizeof(T), Vals);
+
+           Vals = block_load<T, N>(Out + ElemOff);
+           Vals += 2;
+           auto View = Vals.template select<N, 1>();
+           if constexpr (UseProperties)
+             block_store<T, N>(Out, ElemOff * sizeof(T), View,
+                               StorePropertiesT{});
+           else
+             block_store<T, N>(Out, ElemOff * sizeof(T), View);
+
+           Vals = block_load<T, N>(Out + ElemOff);
+           Vals += 3;
+           View = Vals.template select<N, 1>();
+           if constexpr (UseProperties)
+             block_store<T, N>(Out + ElemOff, View, StorePropertiesT{});
+           else
+             block_store<T, N>(Out + ElemOff, View);
+         }
+       });
+     }).wait();
+  } catch (sycl::exception const &e) {
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+    sycl::free(Out, Q);
+    return false;
+  }
+
+  bool Passed = true;
+
+  for (int i = 0; i < Size; i++) {
+    bool IsMaskSet = (i / N + 1) % 1;
+    Tuint Expected = sycl::bit_cast<Tuint>(Out_val);
+    if (!UseMask || IsMaskSet)
+      Expected = sycl::bit_cast<Tuint>((T)(i + 6));
+    Tuint Computed = sycl::bit_cast<Tuint>(Out[i]);
+    if (Computed != Expected) {
+      Passed = false;
+      std::cout << "Out[" << i << "] = " << std::to_string(Computed) << " vs "
+                << std::to_string(Expected) << std::endl;
+    }
+  }
+
+  sycl::free(Out, Q);
+
+  return Passed;
+}
+
+template <typename T, bool TestPVCFeatures> bool test_block_store(queue Q) {
+  constexpr bool CheckMask = true;
+  constexpr bool CheckProperties = true;
+  properties AlignOnlyProps{alignment<sizeof(T)>};
+
+  bool Passed = true;
+
+  // Test block_store() that is available on Gen12 and PVC.
+  Passed &= testUSM<T, 1, !CheckMask, CheckProperties>(Q, 2, 4, AlignOnlyProps);
+  Passed &= testUSM<T, 2, !CheckMask, CheckProperties>(Q, 1, 4, AlignOnlyProps);
+  Passed &= testUSM<T, 3, !CheckMask, CheckProperties>(Q, 2, 8, AlignOnlyProps);
+  Passed &= testUSM<T, 4, !CheckMask, CheckProperties>(Q, 2, 4, AlignOnlyProps);
+  Passed &= testUSM<T, 8, !CheckMask, CheckProperties>(Q, 2, 4, AlignOnlyProps);
+  Passed &=
+      testUSM<T, 16, !CheckMask, CheckProperties>(Q, 2, 4, AlignOnlyProps);
+  Passed &=
+      testUSM<T, 32, !CheckMask, CheckProperties>(Q, 2, 4, AlignOnlyProps);
+  // Intentionally check non-power-of-2 simd size - it must work.
+  Passed &=
+      testUSM<T, 33, !CheckMask, CheckProperties>(Q, 2, 4, AlignOnlyProps);
+  // TODO: Enable after failure fixed
+  // Passed &=
+  //    testUSM<T, 67, !CheckMask, CheckProperties>(Q, 1, 4, AlignOnlyProps);
```

Comment:
This test case fails even when using the old API. I wanted to see what happens before we moved to intrinsics, but we actually assert that the size is a multiple of 16, so we couldn't do it in the old way. I made an internal tracker for this.

Reply:
That was my guess too but I did not have enough courage to say it :)

Reply:
We still need to analyse/check it on our side first.

Reply:
Of course, I have an internal tracker assigned to me for this.